src/bosh-template/lib/bosh/template/evaluation_context.rb

```diff
@@ -40,11 +40,7 @@ def initialize(spec, dns_encoder)
                     "Hash expected, #{spec.class} given"
         end
 
-        if spec['job'].is_a?(Hash)
-          @name = spec['job']['name']
-        else
-          @name = nil
-        end
+        @name = spec['name']
```

Comment:
Curious about this change, is this always safe, even if a v1 manifest is being consumed?

Reply:
Considering that “job” means “instance group”, if the instance group is not a Hash then what could it be? An array? `nil`? Is this something that can happen?

To my knowledge, from within an ERB file that needs being rendered, the instance group _must_ be a Hash. If it’s not, then something bad is happening and then the code should crash.

Basically I’ve applied the “_let it crash_” principle: no reason to protect against a situation that should never happen, and possibly maintain unnecessary code or hide a bug.